src/ORM/Table.php

```diff
@@ -2361,21 +2365,28 @@ public function hasFinder(string $type): bool
      *
      * @param string $type name of the finder to be called
      * @param \Cake\ORM\Query $query The query object to apply the finder options to
-     * @param array $options List of options to pass to the finder
+     * @param array $args List of options to pass to the finder
      * @return \Cake\ORM\Query
      * @throws \BadMethodCallException
      */
-    public function callFinder(string $type, Query $query, array $options = []): Query
+    public function callFinder(string $type, Query $query, ...$args): Query
     {
-        $query->applyOptions($options);
-        $options = $query->getOptions();
+        if ($args) {
```

Comment:
Perhaps use `if (isset($args[0]) && is_array($args[0])) {` and avoid the nested `if` below?